Userland/Applications/Magnifier/MagnifierWidget.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021, Valtteri Koskivuori <vkoskiv@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "MagnifierWidget.h"
+#include <LibGUI/Painter.h>
+#include <LibGUI/Window.h>
+#include <LibGUI/WindowServerConnection.h>
+#include <LibGfx/AffineTransform.h>
+#include <math.h>
+
+MagnifierWidget::MagnifierWidget()
+{
+}
+
+MagnifierWidget::~MagnifierWidget()
+{
+}
+
+void MagnifierWidget::track_cursor_globally()
+{
+    VERIFY(window());
+    auto window_id = window()->window_id();
+    VERIFY(window_id >= 0);
+
+    set_global_cursor_tracking(true);
+    GUI::WindowServerConnection::the().set_global_cursor_tracking(window_id, true);
+}
+
+void MagnifierWidget::set_scaling(int scale)
+{
+    VERIFY(scale == 2 || scale == 4);
+    m_scale_factor = scale;
+    update();
+}
```

Comment:
This feels a bit weird as it uses three different names for the same thing (`scale`, `scaling`, `scale_factor`). Suggestion:

```suggestion
void MagnifierWidget::set_scale_factor(int scale_factor)
{
    VERIFY(scale_factor == 2 || scale_factor == 4);
    m_scale_factor = scale_factor;
    update();
}
```